src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs

```diff
@@ -264,10 +270,70 @@ private int LineBreaksAfter(SyntaxToken currentToken, SyntaxToken nextToken)
             return 0;
         }
 
+        private static bool All<TNode>(SyntaxList<TNode> items, Func<TNode, bool> predicate)
+            where TNode : SyntaxNode
+        {
+            bool result = true;
+            for (int i = 0; i < items.Count; i++)
```

Comment:
>for (int i = 0; i < items.Count; i++) [](start = 12, length = 37)

It feels like we should be able to use foreach here.

Reply:
The reason for a standard "for" is perf, but since you prefer the foreach, I am changing this.

Reply:
> The reason for a standard "for" is perf, but since you prefer the foreach, I am changing this.

I think SyntaxList has custom struct-based enumerator, which should avoid extra allocations and interface calls during foreach.

---
In reply to: [530998463](https://github.com/dotnet/roslyn/pull/49495#discussion_r530998463) [](ancestors = 530998463)

Reply:
I've seen the stack based implementation. You will agree with me the simple iteration has less overhead. Anyway, I did change the implementation with foreach.

Reply:
> I've seen the stack based implementation. 

What exactly are you referring to? Could you please elaborate?

---
In reply to: [531231606](https://github.com/dotnet/roslyn/pull/49495#discussion_r531231606) [](ancestors = 531231606)

Reply:
I meant that a simple for is anyway more performant than an enumerable even if it has been optimized for value types.
But nevermind, let's go on.